System test failure with K8s v1.6.x

I am building and testing k8s-policy v0.6.0 from source. Because I am trying to use Calico with K8s v1.6.x, so i have been testing the policy-control with k8s v1.6.x image. 

The following is the code change to Makefile

```diff
--- a/Makefile
+++ b/Makefile

@@ -21,7 +21,7 @@ st: docker-image run-etcd run-k8s-apiserver
        $(MAKE) stop-k8s-apiserver stop-etcd
 GET_CONTAINER_IP := docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}'
-K8S_VERSION=1.5.3
+K8S_VERSION=1.6.3
```

Then issuing `make st`, I got following error
```
echo "VERSION='`git describe --tags --dirty`'" > version.py
# Build the docker image for the policy controller.
touch image.created
st-etcd
docker run --detach \
--name st-etcd quay.io/coreos/etcd:v3.1.5 \
etcd \
--advertise-client-urls "http://127.0.0.1:2379,http://127.0.0.1:4001" \
--listen-client-urls "http://0.0.0.0:2379,http://0.0.0.0:4001"
af6340b86869dd6abd8785a3c5bc680b50f971ea4402cc055d563b5d363f455c
st-apiserver
ETCD_IP=`docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' st-etcd` && 
docker run --detach \
  --name st-apiserver \
gcr.io/google_containers/hyperkube-amd64:v1.6.3 \
          /hyperkube apiserver --etcd-servers=http://${ETCD_IP}:2379 \
          --service-cluster-ip-range=10.101.0.0/16 -v=10
af5fcbed6f58166e096cd4698482ce12dadac0f782c2ef3370910bc4c9ef897d
./tests/system/apiserver-reconnection.sh
++ get_container_ip st-etcd
++ docker inspect '--format={{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' st-etcd
+ ETCD_IP=172.17.0.2
++ get_container_ip st-apiserver
++ docker inspect '--format={{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' st-apiserver
+ K8S_IP=172.17.0.3
+ docker rm -f calico-policy-controller
calico-policy-controller
+ sleep 2
+ docker run --detach --name=calico-policy-controller -e K8S_API=https://172.17.0.3:6443 -e K8S_INSECURE_SKIP_TLS_VERIFY=true -e ETCD_ENDPOINTS=http://172.17.0.2:2379 calico/kube-policy-controller
ea185f21d240390afc2481d1adae15063ed827613df29043da5f87b2a0e69127
+ sleep 2
+ NS_NAME=chocolate
+ create_namespace chocolate
+ name=chocolate
+ curl -k -H 'Content-Type: application/yaml' -XPOST --data-binary @- https://172.17.0.3:6443/api/v1/namespaces
Unauthorized
+ sleep 1
+ docker exec st-etcd etcdctl ls --recursive /calico
+ grep chocolate
Makefile:20: recipe for target 'st' failed
make: *** [st] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ st: docker-image run-etcd run-k8s-apiserver
 	$(MAKE) stop-k8s-apiserver stop-etcd
 
 GET_CONTAINER_IP := docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}'
-K8S_VERSION=1.5.3
+K8S_VERSION=1.6.4
 .PHONY: run-k8s-apiserver stop-k8s-apiserver run-etcd stop-etcd
 run-k8s-apiserver: stop-k8s-apiserver
 	ETCD_IP=`$(GET_CONTAINER_IP) st-etcd` && \
@@ -29,7 +29,8 @@ run-k8s-apiserver: stop-k8s-apiserver
 	  --name st-apiserver \
 	gcr.io/google_containers/hyperkube-amd64:v$(K8S_VERSION) \
 		  /hyperkube apiserver --etcd-servers=http://$${ETCD_IP}:2379 \
-		  --service-cluster-ip-range=10.101.0.0/16 -v=10
+		  --service-cluster-ip-range=10.101.0.0/16 -v=10 \
+		  --authorization-mode=RBAC
 
 stop-k8s-apiserver:
 	@-docker rm -f st-apiserver

--- a/tests/system/apiserver-reconnection.sh
+++ b/tests/system/apiserver-reconnection.sh
@@ -1,13 +1,19 @@
 #!/bin/bash -ex
 
-# Utilities.
 function get_container_ip {
     docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $1
 }
 
+ETCD_IP=`get_container_ip st-etcd`
+K8S_IP=`get_container_ip st-apiserver`
+
+while ! docker exec st-apiserver kubectl create clusterrolebinding anonymous-admin --clusterrole=cluster-admin --user=system:anonymous; do
+    sleep 2
+done
+
 function create_namespace {
     name=$1
-    curl -k -H "Content-Type: application/yaml" -XPOST --data-binary @- https://172.17.0.3:6443/api/v1/namespaces <<EOF
+    curl -k -H "Content-Type: application/yaml" -XPOST --data-binary @- https://${K8S_IP}:6443/api/v1/namespaces <<EOF
 apiVersion: v1
 kind: Namespace
 metadata:
@@ -15,9 +21,6 @@ metadata:
 EOF
     sleep 1
 }
-
-ETCD_IP=`get_container_ip st-etcd`
-K8S_IP=`get_container_ip st-apiserver`
 
 # Run policy controller.
 docker rm -f calico-policy-controller || true

